clean_silva.py

```diff
@@ -68,84 +92,89 @@ def writetab(tab, handle):
 		line = key+"\t"+value+"\n"
 		out.write(line)
 
-def cleantab(tab, aux):
-	output = dict()
-	with open(tab) as csvfile:
-		reader = csv.reader(csvfile, delimiter="\t")
-		for row in reader:
-			ID = row[0]
-			tax = row[1]
-			#print(tax)
-			prok = re.search("^Archaea;|^Bacteria;", tax)
-			if(prok != None):
-				if(tax in aux):
-					tax = aux[tax]
-				else:
-					tax = tax.replace("'", "")
-					tax = tax.replace("]", "")
-					tax = tax.replace("[", "")
-					#print(tax)
-					##First search for very frequently misasigned clades
-					search = re.search("((Lactob|B)acillus|Streptococcus)_", tax)
-					if(search != None):
-						search = re.search("Bacilli", tax)
-						if(search == None):
-							tax = ";".join(tax.split(";")[0:6])
-					else:
-						search = re.search("(Vibrio_)", tax)
-						if(search != None):
-							search = re.search("Vibrionales", tax)
-							if(search == None):
-								tax = ";".join(tax.split(";")[0:6])
-						else:
-							search = re.search("(Clostridium_)", tax)
-							if(search != None):
-								search = re.search("Clostridiales", tax)
-								if(search == None):
-									tax = ";".join(tax.split(";")[0:6])
-							else:
-								search = re.search("(Mycobacterium_)", tax)
-								if(search != None):
-									search = re.search("Corynebacteriales", tax)
-									if(search == None):
-										tax = ";".join(tax.split(";")[0:6])
-								else:
-									search = re.search("(Pseudomonas_)", tax)
-									if(search != None):
-										search = re.search("Pseudomonadales", tax)
-										if(search == None):
-											tax = ";".join(tax.split(";")[0:6])
-					#print(tax)
-					prob = re.search(isproblem, tax)
-					if(prob != None):
-						for item in keywords:
-							search = re.search(item, tax)
-							if(search != None):
-								tax = tax.split(item)[0]
-						#print("Got a problem")
-						marker=re.compile(";[A-Z][a-z]+(eae|ales|proteobacteria)_(str|sp|pv|subsp|genomosp|gen)\.")
-						prob = re.search(marker, tax)
-						if(prob != None):
-							tax = ";".join(tax.split(";")[0:6])
-						else:
-							marker=re.compile(";[A-Z][a-z]+(eae|ales|proteobacteria|archaeota)_(bacterium|archaeon)")
-							prob = re.search(marker, tax)
-							if(prob != None):
-								tax = ";".join(tax.split(";")[0:6])
-							else:
-								tax = re.sub("(_subsp\._|_genomosp\._|_sp\._|_str\._|_pv\._|_aff\._|cf\._|_gen\._)", ";", tax)
-								tax = re.sub("_sp\.", ";", tax)
-					search = re.search("Candidatus_[A-Za-z]+_[A-Za-z]+_", tax)
-					if(search != None):
-						tax=re.sub("Candidatus_[A-Za-z]+_[A-Za-z]+(_)", ";", tax)
-					else:
-						search = re.search(";[A-Za-z]+_[A-Za-z]+_", tax)
-						if(search != None):
-							tax=re.sub("[A-Za-z]+_[A-Za-z]+(_)", ";", tax)
-					#print(tax)
-				tax = tax.split(";;")[0]
-			output[ID] = tax
-	return(output)
+
+def process_row(aux, row):
```

Comment:
I had to put the row processing into a separate function in order to send the rows in parallel to the multiprocessing worker pool.